src/workers.ts

```diff
@@ -143,40 +85,53 @@ export function getWorkerLocation(): WorkerLocation {
   };
 }
 
+/** Re-dispatch progress events */
+function reDispatchProgressEvent(event: ProgressEmit): void {
+  // eslint-disable-next-line no-restricted-globals
+  self.dispatchEvent(event);
+}
+
 const workers: Partial<PenumbraWorkers> = {};
 
 /** Instantiate a Penumbra Worker */
-export function createPenumbraWorker(url: URL | string): PenumbraWorker {
+export async function createPenumbraWorker(
+  url: URL | string,
+): Promise<PenumbraWorker> {
   // Use string literals to provide default worker URL hints to webpack
-  switch (String(url)) {
-    // case DEFAULT_WORKERS.decrypt: {
-    //   const worker = new Worker('decrypt.penumbra.worker.js', {
-    //     type: 'module',
-    //   });
-    //   return { worker, comlink: Comlink.wrap(worker) };
-    // }
-    // case DEFAULT_WORKERS.zip: {
-    //   const worker = new Worker('zip.penumbra.worker.js', { type: 'module' });
-    //   return { worker, comlink: Comlink.wrap(worker) };
-    // }
-    // case DEFAULT_WORKERS.StreamSaver: {
-    //   const worker = new Worker('./streamsaver.penumbra.serviceworker.js', { type: 'module' });
-    //   return { worker, comlink: Comlink.wrap(worker) };
-    // }
-    default: {
-      const worker = new Worker(url, { type: 'module' });
-      return { worker, comlink: Comlink.wrap(worker) };
-    }
-  }
+  // switch (String(url)) {
```

Comment:
are these comments still needed?

Reply:
I would keep those comments in case someone has issues in the future that requires manual hinting for the compiler.